graph/pull_v2.go

```diff
@@ -117,18 +121,28 @@ type errVerification struct{}
 
 func (errVerification) Error() string { return "verification failed" }
 
+func (p *v2Puller) extractSize(di *downloadInfo) (size int64, err error) {
+	blobs := p.repo.Blobs(context.Background())
+	desc, err := blobs.Stat(context.Background(), di.digest)
+	if err != nil {
+		logrus.Debugf("Error statting layer: %v", err)
+	}
+	return desc.Size, err
+}
+
 func (p *v2Puller) download(di *downloadInfo) {
 	logrus.Debugf("pulling blob %q to %s", di.digest, di.img.ID)
 
 	blobs := p.repo.Blobs(context.Background())
 
-	desc, err := blobs.Stat(context.Background(), di.digest)
-	if err != nil {
-		logrus.Debugf("Error statting layer: %v", err)
-		di.err <- err
-		return
+	if di.size == 0 {
+		size, err := p.extractSize(di)
+		di.size = size
+		if err != nil {
+			di.err <- err
+			return
+		}
```

Comment:
I don't think it makes a difference here, but in general it's best to check the error before using other return values.